src/pages/single-video-page/single-video-page.js

```diff
@@ -0,0 +1,26 @@
+import { useParams } from "react-router-dom";
+import { MyYouTube } from "../../components";
+import { useState, useEffect } from "react";
+import { getData, useData } from "../../helpers";
+
+export function SingleVideoPage() {
+  const { id: video_id } = useParams();
+  const { videos, setVideos } = useData();
+  const [video, setVideo] = useState(null);
+  useEffect(() => {
+    if (videos.length === 0)
+      (async () => {
```

Comment:
I would suggest try making separate async function and then call inside useffect.

Reply:
Yeah, that would be good for bigger functions as I will have to pass it as a dependency array too. Will keep it in mind if the function's size increases.